packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -146,20 +148,69 @@ class MDnsObservatoryDiscovery {
       applicationId: applicationId,
       deviceVmservicePort: deviceVmservicePort,
     );
-    Uri observatoryUri;
-    if (result != null) {
-      final String host = usesIpv6
-        ? InternetAddress.loopbackIPv6.address
-        : InternetAddress.loopbackIPv4.address;
-      observatoryUri = await buildObservatoryUri(
-        device,
-        host,
-        result.port,
-        hostVmservicePort,
-        result.authCode,
-      );
+    if (result == null) {
```

Comment:
Should we only be checking this on macOS?

Reply:
If the lookup fails, I thought it might be useful to get the verbose tracing that `_checkForIPv4LinkLocal()` produces regardless of host/target OS. Whether to print a louder error with advice is then gated on the device type. In this case it's iOS, which is limited to macOS. If we know better advice for other host/target combos in the future, we can add them there.

Reply:
SGTM